titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/endpoint/v3/grpc/DefaultJobManagementServiceGrpc.java

```diff
@@ -602,6 +615,12 @@ public void deleteTaskAttributes(TaskAttributesDeleteRequest request, StreamObse
 
     @Override
     public void observeJobs(ObserveJobsQuery query, StreamObserver<JobChangeNotification> responseObserver) {
+        Stopwatch start = Stopwatch.createStarted();
+
+        String trxId = UUID.randomUUID().toString();
```

Comment:
Do you see value in using thread ID as `trxId` in case we want to correlated it with observations metrics/logs.

Reply:
We do not log this value, only use as a unique id to correlate metrics.